apis/src/main/kotlin/org/yapp/apis/auth/helper/google/GoogleIdTokenProcessor.kt

```
@@ -0,0 +1,34 @@
+        GsonFactory.getDefaultInstance()
+    )
+        .setAudience(Collections.singletonList(googleOauthProperties.clientId))
+        .build()
+
+    fun parseAndValidate(idToken: String): GoogleIdToken.Payload {
+        try {
+            val googleIdToken = verifier.verify(idToken)
+                ?: throw AuthException(AuthErrorCode.INVALID_GOOGLE_ID_TOKEN, "Invalid ID token")
+
+            return googleIdToken.payload
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)